django/core/management/commands/loaddata.py

```diff
@@ -156,63 +165,75 @@ def loaddata(self, fixture_labels):
                     % (self.loaded_object_count, self.fixture_object_count, self.fixture_count)
                 )
 
+    def _store_obj(self, obj):
+        """Store an object if permitted.
+
+        Conditions of permission are that the object's app isn't excluded and
+        and that the router allows migrations for the object's model. Return
+        whether those conditions were met.
+        """
+        if (obj.object._meta.app_config in self.excluded_apps or
+                type(obj.object) in self.excluded_models):
+            return False
+        stored = False
+        if router.allow_migrate_model(self.using, obj.object.__class__):
+            stored = True
+            self.models.add(obj.object.__class__)
+            try:
+                obj.save(using=self.using)
+            # psycopg2 raises ValueError if data contains NUL chars.
+            except (DatabaseError, IntegrityError, ValueError) as e:
+                e.args = ("Could not load %(object_label)s(pk=%(pk)s): %(error_msg)s" % {
+                    'object_label': obj.object._meta.label,
+                    'pk': obj.object.pk,
+                    'error_msg': e,
+                },)
+                raise
+        if obj.deferred_fields:
+            self.objs_with_deferred_fields.append(obj)
+        return stored
+
     def load_label(self, fixture_label):
         """Load fixtures files for a given label."""
         show_progress = self.verbosity >= 3
         for fixture_file, fixture_dir, fixture_name in self.find_fixtures(fixture_label):
-            _, ser_fmt, cmp_fmt = self.parse_name(os.path.basename(fixture_file))
```

Comment:
IMO we should revert this change `fixture_file` is an absolute path, so it can contain directories with dots in names, this can cause issues in `parse_name()`.